modules/EvseV2G/connection.cpp

```diff
@@ -985,3 +1021,51 @@ int connection_start_servers(struct v2g_context* ctx) {
 
     return 0;
 }
+int create_udp_socket(const int udp_port) {
+    // find suitable addresses
+    addrinfo hints;
+    addrinfo *result, *result_pointer;
+    int udp_socket;
+
+    memset(&hints, 0, sizeof(addrinfo));
+    hints.ai_family = AF_UNSPEC;    // allow IPv4 or IPv6
+    hints.ai_socktype = SOCK_DGRAM; // Datagram socket
+    hints.ai_flags = AI_ALL;        // For IPv4 or IPv6
+    hints.ai_protocol = 0;          // OS selects best protocol (best practise)
+
+    // convert port to string
+    char port_str[6];
+    snprintf(port_str, sizeof(port_str), "%d", udp_port);
+
+    //              alternatively:
+    //              "localhost"
+    auto get_addrinfo_result = getaddrinfo(nullptr, port_str, &hints, &result);
+
+    if (get_addrinfo_result != 0) {
+        dlog(DLOG_LEVEL_ERROR, "getaddrinfo: %s", gai_strerror(get_addrinfo_result));
+        return -1;
+    }
+
+    // create the socket
+    for (result_pointer = result; result_pointer != nullptr; result_pointer = result_pointer->ai_next) {
```

Comment:
`result_pointer = result` sounds wrong, better use `initial_addr` and `current_addr` or something similar